src/System.Net.Mail/tests/Functional/SmtpClientTest.cs

```diff
@@ -332,6 +332,82 @@ public async Task TestMailDeliveryAsync()
             }
         }
 
+        [Fact]
+        public void TestSmtpDeliveryFormatSevenBit() {
+            SmtpServer server = null;
+            SmtpClient client = null;
+            var serverList = new System.Collections.Generic.List<SmtpServer>();
+
+            MailMessage msg = new MailMessage("foo@example.com", "bar@example.com");
+            msg.HeadersEncoding = msg.BodyEncoding = msg.SubjectEncoding = System.Text.Encoding.UTF8;
+
+            var subjectTxt = "Test 测试 Contain 包含 UTF8";
+            var subjectEncode = "=?utf-8?B?VGVzdCDmtYvor5UgQ29udGFpbiDljIXlkKsgVVRGOA==?=";
+
+            msg.Subject = subjectTxt;
+
+            Action<SmtpDeliveryFormat, bool, string, string, string> test = (format, utf8, async, sync, tag) => {
+                var retry = 0;
+                while (retry < 3) {
+                    if (client != null) {
+                        server.Stop();
+                        client.Dispose();
+                    }
+                    //New connections are required when parameters change
+                    //How to close cached SmtpConnection !!?? Killing this server is the quickest solution.
+                    server = new SmtpServer();
+                    var running = false;
+                    Thread t = new Thread(() => {
+                        running = true;
+                        server.Run();
+                    });
+                    t.Start();
+                    while (!running) {//wait fist server started
+                        Thread.Sleep(1);
+                    }
+                    Thread.Sleep(100);//wait all server running
+                    server.Support_SMTPUTF8 = utf8;
+
+                    client = new SmtpClient("localhost", server.EndPoint.Port);
+                    client.DeliveryFormat = format;
+
+
+                    var task = client.SendMailAsync(msg);
+                    if (!task.Wait(5000)) {//If the server did not accept the request? Try again.
```

Comment:
Why is there a 5 second delay here?  If a localhost test SMTP server is being used here, it should be close to 100% reliable.  There shouldn't need to be any delays at all.

Reply:
When I run this code, I find that occasionally I get stuck here. Breakpoint finds that the server has not received the request. Maybe the server is not ready.

Now using await, there's no such problem, haha.

The new code style has been optimized. The new test method uses code from the current method `TestMailDeliveryAsync`. But there's one thing about the code that I don't understand. I did not include this line in the test code:

https://github.com/dotnet/corefx/blob/29cfdd9b8e43f65d57c3201486256db87ea118e2/src/System.Net.Mail/tests/Functional/SmtpClientTest.cs#L321